Backend/Backend.Tests/PlaylistTests.cs

```diff
@@ -0,0 +1,354 @@
+    public async void PlaylistDelete_InvalidPlaylistId_ReturnsFalse()
+    {
+        Mock<AppDbContext> dbContextMock = new(new DbContextOptions<AppDbContext>());
+        Mock<IMapper> mapperMock = new();
+        PlaylistService service = new(dbContextMock.Object, mapperMock.Object);
+        User user = new User()
+        {
+            Id = Guid.NewGuid()
+        };
+
+        Guid playlistGuid = Guid.NewGuid();
+        Mock<DbSet<Playlist>> playlists = new[]
+        {
+            new Playlist(dbContextMock.Object)
+            {
+                Id = playlistGuid,
+                UserId = user.Id,
+                Name = "XXXXXXX"
+            },
+        }.AsQueryable().BuildMockDbSet();
+        
+        dbContextMock.SetupGet(db => db.Playlists).Returns(playlists.Object);
+        bool result = await service.Delete(user, Guid.NewGuid());
+        Assert.False(result);
+    }  
+    
+    [Fact]
+    public async void PlaylistDelete_Valid_ReturnsTrue()
+    {
+        Mock<AppDbContext> dbContextMock = new(new DbContextOptions<AppDbContext>());
+        Mock<IMapper> mapperMock = new();
+        PlaylistService service = new(dbContextMock.Object, mapperMock.Object);
+        User user = new User()
+        {
+            Id = Guid.NewGuid()
+        };
+
+        Guid playlistGuid = Guid.NewGuid();
+        Mock<DbSet<Playlist>> playlists = new[]
+        {
+            new Playlist(dbContextMock.Object)
+            {
+                Id = playlistGuid,
+                UserId = user.Id,
+                Name = "XXXXXXX"
+            },
+        }.AsQueryable().BuildMockDbSet();
+        
+        dbContextMock.SetupGet(db => db.Playlists).Returns(playlists.Object);
+        bool result = await service.Delete(user, playlistGuid);
+        Assert.True(result);
+        //Assert.True(playlists.Object.Count(e => true) == 0);
+    }  
+    
+    [Fact]
+    public async void PlaylistDeleteElement_InValid_ReturnsFalse()
+    {
+        Mock<AppDbContext> dbContextMock = new(new DbContextOptions<AppDbContext>());
+        Mock<IMapper> mapperMock = new();
+        PlaylistService service = new(dbContextMock.Object, mapperMock.Object);
+        User user = new User()
+        {
+            Id = Guid.NewGuid()
+        };
+                
+        Mock<DbSet<Playlist>> playlists = new Playlist[]
+        {
+        }.AsQueryable().BuildMockDbSet();
+        Mock<DbSet<PlaylistElement>> elements = new PlaylistElement[]
+        {
+        }.AsQueryable().BuildMockDbSet(); 
+        
+        dbContextMock.SetupGet(db => db.Playlists).Returns(playlists.Object);
+        dbContextMock.SetupGet(db => db.PlaylistElements).Returns(elements.Object);
+
+        bool result = await service.DeleteElement(user, Guid.Empty);
+        Assert.False(result);
+    }  
+    
+    [Fact]
+    public async void PlaylistDeleteElement_Valid_ReturnsTrue()
+    {
+        Mock<AppDbContext> dbContextMock = new(new DbContextOptions<AppDbContext>());
```

Comment:
Theres a lot of code duplication that you couldve exported out to a private setup function. not game breaking, but it would make this way easier on you